main.go

```diff
@@ -626,8 +626,31 @@ func (o *ObjectivesServer) GetMultiBurnrateAlerts(ctx context.Context, expr, gro
 	}
 
 	// Match alerts that at least have one character for the slo name.
-	// TODO: Adjust the query to only query for SLOs that we're really interested in.
 	queryAlerts := `ALERTS{slo=~".+"}`
+
+	if grouping != "" {
+		// If grouping exists we merge those matchers directly into the queryAlerts query.
+		groupingMatchers, err := parser.ParseMetricSelector(grouping)
+		if err != nil {
+			return openapiserver.ImplResponse{}, fmt.Errorf("failed parsing grouping matchers: %w", err)
+		}
+
+		expr, err := parser.ParseExpr(queryAlerts)
+		if err != nil {
+			return openapiserver.ImplResponse{}, fmt.Errorf("failed parsing alerts metric: %w", err)
+		}
+
+		vec := expr.(*parser.VectorSelector)
+		for _, m := range groupingMatchers {
+			if m.Name == labels.MetricName || m.Name == "slo" { // adding some safeguards that shouldn't be allowed.
```

Comment:
This could be part of a validation func `func isValidMatcherName(name string) bool`. But it's also fine for now.

Reply:
If these were more than just the two I'd definitely start putting them in a func, yes. Good point!